### ISSUE TITLE
Show error for Pointer type mismatch at ASR level

### ISSUE THREAD

**Issue by Smit-create** [lcompilers/lpython#694]
Python code:
```py
from ltypes import pointer, i16, Pointer, i32

def f():
    yptr1: Pointer[i16[:]]
    y: i32[2]
    y[0] = 1
    y[1] = 2
    yptr1 = pointer(y)

f()
```

This should raise an error at ASR level. https://github.com/lcompilers/lpython/pull/688#discussion_r906671577
cc @czgdp1807

**Comment by certik** [comment]
Yes. All errors should be raised in AST->ASR phase. The only errors in ASR->LLVM should be "not implemented", nothing else.
